### DISCUSSION TITLE
Issue with cronworkflows

### DISCUSSION THREAD

**Discussion by kobejn-jb** [argoproj/argo-workflows#6571] [Q&A]
I'm having strange issue with cronworkflows, I'm not aware of any changes in config, looks like it just started to happen yesterday.

Workflow objects are created on schedule based on cronworkflow objects but pods are not being created and status in not being updated:
```

 Status:
   Finished At:  <nil>
   Started At:   <nil>
 Events:
   Type    Reason           Age   From                 Message
   ----    ------           ----  ----                 -------
   Normal  WorkflowRunning  61s   workflow-controller  Workflow Running
```

Events appear but "Started at" stays empty

Does anybody have a clue what can be causing that?

I'm not creating bug report because I can't reproduce this on other environments.

I'm running version 3.0.7

**Comment by terrytangyuan** [comment]
Could you provide controller pod’s logs?

**Comment by kobejn-jb** [comment]
Here, ale the logs:
https://drive.google.com/file/d/1z24lWV67QOZwlgGiN6QEoNhRoCTQ7Avg/view?usp=sharing

I haven't found anything helpful there, sometime this message appears:
```
err="workflow never reconciled: copy-parquets-to-gsm-db-cron-1629370800" instanceID=argo-controller labelSelector="!workflows.argoproj.io/phase,workflows.argoproj.io/controller-instanceid=argo-controller" managedNamespace=argo
```

I've also found this message in kubeapi logs:
```
E0820 08:23:42.131897       1 status.go:71] apiserver received an error that is not an metav1.Status: &net.OpError{Op:"write", Net:"tcp", Source:(*net.TCPAddr)(0xc09970b200), Addr:(*net.TCPAddr)(0xc09970b230), Err:(*os.SyscallError)(0xc085c47160)}
E0820 08:23:42.137381       1 writers.go:120] apiserver was unable to write a fallback JSON response: write tcp 10.200.100.131:443->10.200.78.87:37864: write: connection reset by peer
```

It seems to be happening when controller should be crating pods, but I'm not 100% sure if this is related

Also situation right now is that only some cronworkflows goes into that state the rest work fine, also I've notices that it also happens to some workflows (not cronworkflows) but this case is rare.

Can this all be caused by high load on api server? My cluster is EKS 1.18.9

EDIT:
I've noticed controller crashing after that message:
```
time="2021-08-20T09:51:36.083Z" level=fatal msg="Failed to update config: upper: giving up trying to connect: too many clients"
```

**Reply by terrytangyuan** [reply]
Looks like your workflow never reconciled because the pod has failing liveness probe for the last 5 mins: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request

This usually indicates something wrong with your k8s apiserver connection under certain high load.

**Reply by kobejn-jb** [reply]
Did you mean controller pod?

**Comment by kobejn-jb** [comment]
I couldn't find any evidence that there is increased load on my api server, all metrics look similar like last couple of weeks. I've upgraded EKS to version 1.19. Right now it seems to be working but every time I see "deadline exceeded" in cronworkflows:
``` Events:                                                                                                                                                                                                         
   Type     Reason                 Age    From                 Message                                                                                                                                           
   ----     ------                 ----   ----                 -------                                                                                                                                           
   Normal   WorkflowRunning        4m6s   workflow-controller  Workflow Running                                                                                                                                  
   Warning  WorkflowTimedOut       3m29s  workflow-controller  error in entry template execution: Deadline exceeded                                                                                              
   Normal   WorkflowNodeSucceeded  2m31s  workflow-controller  Succeeded node fouling-analysis-activation-cron-1629895200                                                                                        
   Normal   WorkflowSucceeded      2m31s  workflow-controller  Workflow completed 
```

I'm attaching controller logs with debug enabled, last cronworkflow processed here is fouling-analysis-activation-cron-1629895800:
https://drive.google.com/file/d/1z24lWV67QOZwlgGiN6QEoNhRoCTQ7Avg/view?usp=sharing
